### PR TITLE
Initial playwright setup for e2e tests

### PR DESCRIPTION
**Pull request by yuney-worx4you**
#### Summary
Adding intial setup of Playwright e2e tests for the Legal Hold plugin.

Configurations are identical to the 'todo' plugin.

#### Current issues
Compared to the manual flow, I noticed a few anomalies in the automation flow.

1. When the new user lands at the Legal Hold Plugin settings page, the "Create" table doesn't appear despite the plugin being set to true. This table appears only when the plugin is disabled (set to false) and back to true. The save in between also seem to take a while and can be flaky.
2. When searching for usernames the loading takes a while and times out: Scenario 2 - Admin user can create a new legal hold. Also there aren't appropriate locators to refer to, other than `#react-select-2-input`.
3. Edit and download icons also don't have good locators to reference, other than `div:nth-child(10) > a:nth-child(2)`
4. The downloaded file name isn't currently checked for. I didn't have a good idea how the file name was generated. Do we want to check this?

### CODE REVIEW

**Comment by wiggin77** [comment]
@yuney-worx4you I think to have the playwright test run for CI, a workflow file needs to be added.  See [here](https://github.com/mattermost-community/mattermost-plugin-todo/blob/master/.github/workflows/playwright.yml) from an example.  You can probably use that one as is.

**Comment by yuney-worx4you** [comment]
Yup just added it now! Thanks @wiggin77

**Comment by wiggin77** [comment]
@yuney-worx4you it looks like there is some `mattermost-plugin-todo` specific paths in there that need to be updated for this plugin.

File: e2e/playwright/tests/legalhold_plugin.spec.ts
Comment (saturninoabril) [inline e2e/playwright/tests/legalhold_plugin.spec.ts:85]: Let's have a minimum test for this PR since this is a base setup for Playwright so you'll get a grasp on how we should organize the test and be familiarized with all available APIs/functions in the monorepo (in [support folder](https://github.com/mattermost/mattermost/tree/master/e2e-tests/playwright/support), most importantly the [extended test fixtures](https://github.com/mattermost/mattermost/blob/master/e2e-tests/playwright/support/test_fixture.ts)).

With a minimal test, you'll be able to guarantee test stability by running the test multiple times at different projects ([chrome, firefox and ipad](https://github.com/mattermost/mattermost/blob/master/e2e-tests/playwright/playwright.config.ts#L55-L79)).

It could be:

```javascript
import {test, expect} from '@e2e-support/test_fixture';

test('Admin user is able to access the plugin successfully', async ({pw, pages}) => {
    // # Log in as admin
    const {adminUser} = await pw.initSetup();
    const {page} = await pw.testBrowser.login(adminUser);

    // # Visit system console
    const systemConsolePage = new pages.SystemConsolePage(page);
    await systemConsolePage.goto();
    await systemConsolePage.toBeVisible();

    // # Go to PLUGINS > Legal Hold Plugin
    await systemConsolePage.page.getByRole('link', {name: 'Legal Hold Plugin'}).click();

    // # Enable the plugin
    await page.getByTestId('PluginSettings.PluginStates.com+mattermost+plugin-legal-hold.Enablefalse').check();
    await page.getByTestId('PluginSettings.PluginStates.com+mattermost+plugin-legal-hold.Enabletrue').check();
    await page.getByTestId('saveSetting').click();

    // * Verify that the plugin is active and ready to use
    await expect(page.getByTestId('create')).toHaveCount(2);
});
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @yuney-worx4you, very glad to see this Playwright setup! I left comments/suggestions but feel free to ping for questions/clarifications.

Some notes to keep you going:
1. Include a screenshot/s on your PR that the tests passed locally, like
Screenshot 2024-09-11 at 12 14 04 PM

For the succeeding PRs, I'd suggest to write test cases first and have a review from your team before proceeding with writing code for automated tests. Feel free to share the test cases in any format but I'd recommend doing via PR (ex. https://github.com/mattermost/mattermost-test-management/pull/177)

File: e2e/playwright/setup-environment
Comment (yuney-worx4you) [reply e2e/playwright/setup-environment:14]: Yup can confirm this script runs fine for me

**Comment by yuney-worx4you** [comment]
Hi @saturninoabril thank you so much for your review!
I have made the requested changes but can't run the tests successfully locally.
Looks like it's due to `Error: Cannot find module '@e2e-test.playwright-config'`
How can I fix this error? Would it be because my support folder is currently empty?

File: .gitignore
Comment (wiggin77) [inline .gitignore:1]: We don't want to remove the `dist` directory from this list, or the `.DS_Store` or `.idea` below.

**Comment by yuney-worx4you** [comment]
Test passes locally!
image

File: .nvmrc
Comment (saturninoabril) [inline .nvmrc:1]:
```suggestion
16.13.1

```

**Comment by saturninoabril** [comment]
@yuney-worx4you in CI, if failed of ci/check in Playwright due to formatting issue. You may fix by `npm run prettier:fix`.

File: .github/workflows/playwright.yml
Comment (saturninoabril) [inline .github/workflows/playwright.yml:169]: Please always add new line for end-of-line here and to other files.

File: package-lock.json
Comment (saturninoabril) [inline package-lock.json:6]: Probably accidentally added only. Can be removed.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Looking good @yuney-worx4you! Last few comments and should be all good.

File: package-lock.json
Comment (yuney-worx4you) [reply package-lock.json:6]: I'll remove it in the commit but should it be ignored in the `.gitignore`?

Comment (saturninoabril) [reply package-lock.json:6]: No. It's in the root directory and it might have been added when after accidentally ran npm install instead at the correct directories.

Comment (saturninoabril) [reply package-lock.json:6]: @yuney-worx4you The Playwright pipeline failed with:
```
npm ERR! The `npm ci` command can only install with an existing package-lock.json or
npm ERR! npm-shrinkwrap.json with lockfileVersion >= 1. Run an install with npm@5 or
npm ERR! later to generate a package-lock.json file, then try again.
```

The package.json you deleted was not correct. You should delete `package-lock.json` at the root directory and not the [e2e/playwright/package-lock.json](https://github.com/mattermost/mattermost-plugin-legal-hold/pull/97/commits/8705b47126ea06bc143864863bf68a1d2dd3ba24). Please update and see if the pipeline passed without issue.

File: .gitignore
Comment (saturninoabril) [inline .gitignore:19]: Please add new line for end-of-line. Same comment for:
- e2e/playwright/.eslintignore
- e2e/playwright/.prettierignore
- e2e/playwright/setup-environment

**Review by saturninoabril — CHANGES_REQUESTED** [review]

**Comment by wiggin77** [comment]
@saturninoabril can you help with the remaining error?

**Comment by saturninoabril** [comment]
Now passed the previous error but failed on actual tests. The legal hold plugin is inactive during test and so it needs to activate/enable similar to how it's done [here](https://github.com/mattermost-community/mattermost-plugin-todo/blob/master/e2e/playwright/support/init_test.ts#L38-L56). I'll take a look next, just need to go offline for a while.
Screenshot 2024-09-18 at 6 59 56 AM

**Review by saturninoabril — APPROVED** [review]
Thanks @yuney-worx4you, LGTM now!

Playwright failed due to the test server requiring a license. It's currently set and should work as expected once merged. Note that for forked repo, secrets are not being injected (for security reason) and so it's the reason why it's failing here.

**Comment by yuney-worx4you** [comment]
Cool thanks Saturn!! I don't have write access to the repository so can't merge it myself. Would you be able to either give me access or merge it pleaes? Thank you!

**Comment by saturninoabril** [comment]
All passed now.

Let's wait for another review from @wiggin77.

**Pull request merged**
